cpp/src/io/parquet/reader_impl_helpers.hpp

```diff
@@ -115,6 +115,8 @@ class aggregate_reader_metadata {
 
   [[nodiscard]] auto const& get_key_value_metadata() const { return keyval_maps; }
 
+  [[nodiscard]] auto&& get_key_value_metadata() { return std::move(keyval_maps); }
```

Comment:
Looks like there's a bit missing here; I believe this is the right overload set:
```suggestion
  [[nodiscard]] auto const& get_key_value_metadata() const& { return keyval_maps; }

  [[nodiscard]] auto&& get_key_value_metadata() && { return std::move(keyval_maps); }
```
First overload works on any lvalue ref (const or not), while the second one only works for rvalue, as it rips out the keyval_maps object out of the aggregate_reader_metadata object.

Reply:
Great suggestion. Learning to use ref-qualifiers here.